### ISSUE TITLE
Invalid tag name "extension 'node' property 'npmVersion'": Tags may not have any characters that encodeURIComponent encodes.

### ISSUE THREAD

**Issue by vwiencek** [jhipster/generator-jhipster#13746]
##### **Overview of the issue**

When I start command `gradlew npm_run_test -PnodeInstall --no-daemon` (from gitlab CI) I get the following error:

```
2 info using npm@6.14.8
3 info using node@v14.15.0
4 verbose npm-session 11f84155ce227e2d
5 silly install loadCurrentTree
6 silly install readGlobalPackageData
7 timing stage:rollbackFailedOptional Completed in 1ms
8 timing stage:runTopLevelLifecycles Completed in 6ms
9 verbose stack Error: Invalid tag name "extension 'node' property 'npmVersion'": Tags may not have any characters that encodeURIComponent encodes.
9 verbose stack     at invalidTagName (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:91:15)
9 verbose stack     at fromRegistry (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:296:13)
9 verbose stack     at resolve (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:81:12)
9 verbose stack     at npa (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\npm-package-arg\npa.js:52:10)
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install\deps.js:231:14
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:52:35
9 verbose stack     at Array.forEach (<anonymous>)
9 verbose stack     at C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:52:11
9 verbose stack     at Array.forEach (<anonymous>)
9 verbose stack     at asyncMap (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\async-map.js:51:8)
9 verbose stack     at exports.getAllMetadata (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install\deps.js:228:3)
9 verbose stack     at Installer.loadArgMetadata (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install.js:366:3)
9 verbose stack     at Installer.readGlobalPackageData (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\lib\install.js:667:8)
9 verbose stack     at Array.<anonymous> (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\bind-actor.js:15:8)
9 verbose stack     at LOOP (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\chain.js:15:14)
9 verbose stack     at chain (C:\Users\vwien\new\.gradle\nodejs\node-v14.15.0-win-x64\node_modules\npm\node_modules\slide\lib\chain.js:20:5)
```
##### **Reproduce the error**

start command `gradlew npm_run_test -PnodeInstall --no-daemon`

##### **JHipster Version(s)**
7.0.0-beta.1

##### **JHipster configuration**
```
{
  "generator-jhipster": {
    "blueprints": [],
    "otherModules": [],
    "applicationType": "monolith",
    "baseName": "new",
    "jhipsterVersion": "7.0.0-beta.1",
    "skipClient": false,
    "skipServer": false,
    "skipUserManagement": true,
    "skipCheckLengthOfIdentifier": false,
    "skipFakeData": false,
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "testFrameworks": [],
    "pages": [],
    "creationTimestamp": 1611734632808,
    "promptValues": {
      "packageName": "com.mycompany.myapp",
      "nativeLanguage": "en"
    },
    "serviceDiscoveryType": "eureka",
    "reactive": false,
    "authenticationType": "oauth2",
    "packageName": "com.mycompany.myapp",
    "serverPort": "8080",
    "cacheProvider": "no",
    "enableHibernateCache": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "postgresql",
    "buildTool": "gradle",
    "serverSideOptions": [],
    "websocket": false,
    "searchEngine": false,
    "messageBroker": false,
    "enableSwaggerCodegen": false,
    "clientFramework": "vue",
    "withAdminUi": false,
    "clientTheme": "none",
    "enableTranslation": false,
    "nativeLanguage": "en",
    "packageFolder": "com/mycompany/myapp",
    "clientPackageManager": "npm",
    "clientThemeVariant": "",
    "languages": ["en", "fr"]
  }
}

```

##### **Browsers and Operating System**

Windows 10
- [X] Checking this box is mandatory (this is just to show you read everything)

**Comment by atomfrede** [comment]
Can you post the your `build.gradle`?

**Comment by vwiencek** [comment]
sure ; it's the one generated without modifications

```
buildscript {
    repositories {
        mavenLocal()
        mavenCentral()
        gradlePluginPortal()
        maven { url "https://repo.spring.io/plugins-release" }
    }
    dependencies {
        //jhipster-needle-gradle-buildscript-dependency - JHipster will add additional gradle build script plugins here
    }
}

plugins {
    id "java"
    id "maven-publish"
    id "idea"
    id "jacoco"
    id "org.springframework.boot"
    id "com.google.cloud.tools.jib"
    id "com.gorylenko.gradle-git-properties"
    id "com.github.node-gradle.node"
    id "net.ltgt.apt-eclipse"
    id "net.ltgt.apt-idea"
    id "net.ltgt.apt"
    id "org.liquibase.gradle"
    id "org.sonarqube"
    id "io.spring.nohttp"
    //jhipster-needle-gradle-plugins - JHipster will add additional gradle plugins here
}

group = "com.mycompany.myapp"
version = "0.0.1-SNAPSHOT"

description = ""

sourceCompatibility=11
targetCompatibility=11
assert System.properties["java.specification.version"] == "1.8" || "1.9" || "10" || "11" || "12" || "13" || "14" || "15"

apply from: "gradle/docker.gradle"
apply from: "gradle/sonar.gradle"
//jhipster-needle-gradle-apply-from - JHipster will add additional gradle scripts to be applied here

if (project.hasProperty("prod") || project.hasProperty("gae")) {
    apply from: "gradle/profile_prod.gradle"
} else {
    apply from: "gradle/profile_dev.gradle"
}

if (project.hasProperty("war")) {
    apply from: "gradle/war.gradle"
}

if (project.hasProperty("gae")) {
    apply plugin: 'maven'
    apply plugin: 'org.springframework.boot.experimental.thin-launcher'
    apply plugin: 'io.spring.dependency-management'

    dependencyManagement {
        imports {
            mavenBom "tech.jhipster:jhipster-dependencies:${jhipsterDependenciesVersion}"
        }
    }
    appengineStage.dependsOn thinResolve
}

if (project.hasProperty("zipkin")) {
    apply from: "gradle/zipkin.gradle"
}

idea {
    module {
        excludeDirs += files("node_modules")
    }
}

eclipse {
    sourceSets {
        main {
            java {
                srcDirs += ["build/generated/sources/annotationProcessor/java/main"]
            }
        }
    }
}

defaultTasks "bootRun"

springBoot {
    mainClassName = "com.mycompany.myapp.NewApp"
}

test {
    useJUnitPlatform()
    exclude "**/*IT*", "**/*IntTest*"

    testLogging {
        events 'FAILED', 'SKIPPED'
    }
    // uncomment if the tests reports are not generated
    // see https://github.com/jhipster/generator-jhipster/pull/2771 and https://github.com/jhipster/generator-jhipster/pull/4484
    // ignoreFailures true
    reports.html.enabled = false
}

task integrationTest(type: Test) {
    useJUnitPlatform()
    description = "Execute integration tests."
    group = "verification"
    include "**/*IT*", "**/*IntTest*"

    testLogging {
        events 'FAILED', 'SKIPPED'
    }

    if (project.hasProperty('testcontainers')) {
        environment 'spring.profiles.active', 'testcontainers'
    }

    // uncomment if the tests reports are not generated
    // see https://github.com/jhipster/generator-jhipster/pull/2771 and https://github.com/jhipster/generator-jhipster/pull/4484
    // ignoreFailures true
    reports.html.enabled = false
}

check.dependsOn integrationTest
task testReport(type: TestReport) {
    destinationDir = file("$buildDir/reports/tests")
    reportOn test
}

task integrationTestReport(type: TestReport) {
    destinationDir = file("$buildDir/reports/tests")
    reportOn integrationTest
}

if (!project.hasProperty("runList")) {
    project.ext.runList = "main"
}

project.ext.diffChangelogFile = "src/main/resources/config/liquibase/changelog/" + new Date().format("yyyyMMddHHmmss") + "_changelog.xml"

liquibase {
    activities {
        main {
            driver "org.h2.Driver"
            url "jdbc:h2:file:./build/h2db/db/new"
            username "new"
            password ""
            changeLogFile "src/main/resources/config/liquibase/master.xml"
            defaultSchemaName ""
            logLevel "debug"
            classpath "src/main/resources/"
        }
        diffLog {
            driver "org.h2.Driver"
            url "jdbc:h2:file:./build/h2db/db/new"
            username "new"
            password ""
            changeLogFile project.ext.diffChangelogFile
            referenceUrl "hibernate:spring:com.mycompany.myapp.domain?dialect=org.hibernate.dialect.H2Dialect&hibernate.physical_naming_strategy=org.springframework.boot.orm.jpa.hibernate.SpringPhysicalNamingStrategy&hibernate.implicit_naming_strategy=org.springframework.boot.orm.jpa.hibernate.SpringImplicitNamingStrategy"
            defaultSchemaName ""
            logLevel "debug"
            classpath "$buildDir/classes/java/main"
            excludeObjects "oauth_access_token, oauth_approvals, oauth_client_details, oauth_client_token, oauth_code, oauth_refresh_token"
        }
    }

    runList = project.ext.runList
}

gitProperties {
    failOnNoGitDirectory = false
    keys = ["git.branch", "git.commit.id.abbrev", "git.commit.id.describe"]
}

checkstyle {
    toolVersion "${checkstyleVersion}"
    configFile file("checkstyle.xml")
    checkstyleTest.enabled = false
}
nohttp {
    source.include "build.gradle", "README.md"
}

configurations {
    providedRuntime
    implementation.exclude module: "spring-boot-starter-tomcat"
    all {
        resolutionStrategy {
            // Inherited version from Spring Boot can't be used because of regressions:
            // To be removed as soon as spring-boot use the same version
            force 'org.liquibase:liquibase-core:4.2.2'
        }
    }
}

repositories {
    mavenLocal()
    mavenCentral()
    jcenter()
    //jhipster-needle-gradle-repositories - JHipster will add additional repositories
}

dependencies {
    // import JHipster dependencies BOM
    if (!project.hasProperty("gae")) {
        implementation platform("tech.jhipster:jhipster-dependencies:${jhipsterDependenciesVersion}")
    }

    // Use ", version: jhipsterDependenciesVersion, changing: true" if you want
    // to use a SNAPSHOT release instead of a stable release
    implementation group: "tech.jhipster", name: "jhipster-framework"
    implementation "javax.annotation:javax.annotation-api"
    implementation "io.dropwizard.metrics:metrics-core"
    implementation "io.micrometer:micrometer-registry-prometheus"
    implementation "com.fasterxml.jackson.datatype:jackson-datatype-hppc"
    implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310"
    implementation "com.fasterxml.jackson.module:jackson-module-jaxb-annotations"
    implementation "com.fasterxml.jackson.datatype:jackson-datatype-hibernate5"
    implementation "com.fasterxml.jackson.core:jackson-annotations"
    implementation "com.fasterxml.jackson.core:jackson-databind"
    implementation "org.hibernate:hibernate-core"
    implementation "com.zaxxer:HikariCP"
    implementation "org.apache.commons:commons-lang3"
    implementation "javax.transaction:javax.transaction-api"
    implementation "org.hibernate:hibernate-entitymanager"
    implementation "org.hibernate.validator:hibernate-validator"
    implementation "org.liquibase:liquibase-core"
    liquibaseRuntime "org.liquibase:liquibase-core"
    liquibaseRuntime "org.liquibase.ext:liquibase-hibernate5:${liquibaseHibernate5Version}"
    liquibaseRuntime sourceSets.main.compileClasspath
    implementation "org.springframework.boot:spring-boot-loader-tools"
    implementation "org.springframework.boot:spring-boot-starter-mail"
    implementation "org.springframework.boot:spring-boot-starter-logging"
    implementation "org.springframework.boot:spring-boot-starter-actuator"
    implementation "org.springframework.boot:spring-boot-starter-data-jpa"
    testImplementation "org.testcontainers:postgresql"
    implementation "org.springframework.boot:spring-boot-starter-security"
    implementation ("org.springframework.boot:spring-boot-starter-web") {
        exclude module: "spring-boot-starter-tomcat"
    }
    implementation "org.springframework.boot:spring-boot-starter-undertow"
    implementation "org.springframework.boot:spring-boot-starter-thymeleaf"
    implementation "org.zalando:problem-spring-web"
    implementation "org.springframework.cloud:spring-cloud-starter-netflix-eureka-client"
    implementation "org.springframework.cloud:spring-cloud-starter-config"
    implementation "org.springframework.security:spring-security-config"
    implementation "org.springframework.security:spring-security-data"
    implementation "org.springframework.security:spring-security-web"
    implementation "org.springframework.boot:spring-boot-starter-oauth2-client"
    implementation "org.springframework.boot:spring-boot-starter-oauth2-resource-server"
    implementation ("io.springfox:springfox-oas")
    implementation ("io.springfox:springfox-swagger2")
    implementation "io.springfox:springfox-bean-validators"
    implementation "org.postgresql:postgresql"
    liquibaseRuntime "org.postgresql:postgresql"
    implementation "org.mapstruct:mapstruct:${mapstructVersion}"
    annotationProcessor "org.mapstruct:mapstruct-processor:${mapstructVersion}"
    annotationProcessor "org.hibernate:hibernate-jpamodelgen:${hibernateVersion}"
    annotationProcessor "org.glassfish.jaxb:jaxb-runtime:${jaxbRuntimeVersion}"
    annotationProcessor "org.springframework.boot:spring-boot-configuration-processor:${springBootVersion}"
    testImplementation ("org.springframework.boot:spring-boot-starter-test") {
        exclude group: "org.junit.vintage", module: "junit-vintage-engine"
    }
    testImplementation "org.springframework.security:spring-security-test"
    testImplementation "org.springframework.boot:spring-boot-test"
    testImplementation "com.tngtech.archunit:archunit-junit5-api:${archunitJunit5Version}"
    testRuntimeOnly "com.tngtech.archunit:archunit-junit5-engine:${archunitJunit5Version}"
    testImplementation "com.h2database:h2"
    liquibaseRuntime "com.h2database:h2"
    //jhipster-needle-gradle-dependency - JHipster will add additional dependencies here
}

if (project.hasProperty("gae")) {
    task createPom {
        def basePath = 'build/resources/main/META-INF/maven'
        doLast {
            pom {
                withXml(dependencyManagement.pomConfigurer)
            }.writeTo("${basePath}/${project.group}/${project.name}/pom.xml")
        }
    }
    bootJar.dependsOn = [createPom]
}

task cleanResources(type: Delete) {
    delete "build/resources"
}

wrapper {
    gradleVersion = "6.8"
}


if (project.hasProperty("nodeInstall")) {
    node {
        version = "${nodeVersion}"
        npmVersion = "${npmVersion}"
        download = true
    }
}
compileJava.dependsOn processResources
processResources.dependsOn bootBuildInfo

```

**Comment by vwiencek** [comment]
I managed to get rid of the error by harcoding the version in the `build.gradle file`

**Comment by vwiencek** [comment]
And when I manually set version numbers within `build.gradle` file, I get the following

```
2 info using npm@6.14.11
3 info using node@v14.15.0
4 verbose run-script [ 'pretest', 'test', 'posttest' ]
5 info lifecycle vsa-dashboard@0.0.0~pretest: vsa-dashboard@0.0.0
6 verbose lifecycle vsa-dashboard@0.0.0~pretest: unsafe-perm in lifecycle true
7 verbose lifecycle vsa-dashboard@0.0.0~pretest: PATH: C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\vwien\dev\workspace\vsa_dashboard\node_modules\.bin;C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11;C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\nodejs\node-v14.15.0-win-x64;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\nodejs\;C:\Users\vwien\AppData\Local\Microsoft\WindowsApps;C:\Users\vwien\AppData\Roaming\npm
8 verbose lifecycle vsa-dashboard@0.0.0~pretest: CWD: C:\Users\vwien\dev\workspace\vsa_dashboard
9 silly lifecycle vsa-dashboard@0.0.0~pretest: Args: [ '/d /s /c', 'npm run lint' ]
10 silly lifecycle vsa-dashboard@0.0.0~pretest: Returned: code: 0  signal: null
11 info lifecycle vsa-dashboard@0.0.0~test: vsa-dashboard@0.0.0
12 verbose lifecycle vsa-dashboard@0.0.0~test: unsafe-perm in lifecycle true
13 verbose lifecycle vsa-dashboard@0.0.0~test: PATH: C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin;C:\Users\vwien\dev\workspace\vsa_dashboard\node_modules\.bin;C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11;C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\nodejs\node-v14.15.0-win-x64;C:\Program Files\Common Files\Oracle\Java\javapath;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0\;C:\Windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\Program Files\NVIDIA Corporation\NVIDIA NvDLISR;C:\Program Files\Docker\Docker\resources\bin;C:\ProgramData\DockerDesktop\version-bin;C:\Program Files\Microsoft VS Code\bin;C:\Program Files\nodejs\;C:\Users\vwien\AppData\Local\Microsoft\WindowsApps;C:\Users\vwien\AppData\Roaming\npm
14 verbose lifecycle vsa-dashboard@0.0.0~test: CWD: C:\Users\vwien\dev\workspace\vsa_dashboard
15 silly lifecycle vsa-dashboard@0.0.0~test: Args: [ '/d /s /c', 'npm run jest' ]
16 silly lifecycle vsa-dashboard@0.0.0~test: Returned: code: 1  signal: null
17 info lifecycle vsa-dashboard@0.0.0~test: Failed to exec test script
18 verbose stack Error: vsa-dashboard@0.0.0 test: `npm run jest`
18 verbose stack Exit status 1
18 verbose stack     at EventEmitter.<anonymous> (C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
18 verbose stack     at EventEmitter.emit (events.js:315:20)
18 verbose stack     at ChildProcess.<anonymous> (C:\Users\vwien\dev\workspace\vsa_dashboard\.gradle\npm\npm-v6.14.11\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
18 verbose stack     at ChildProcess.emit (events.js:315:20)
18 verbose stack     at maybeClose (internal/child_process.js:1048:16)
18 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:288:5)
19 verbose pkgid vsa-dashboard@0.0.0
20 verbose cwd C:\Users\vwien\dev\workspace\vsa_dashboard
21 verbose Windows_NT 10.0.19042
22 verbose argv "C:\\Users\\vwien\\dev\\workspace\\vsa_dashboard\\.gradle\\nodejs\\node-v14.15.0-win-x64\\node.exe" "C:\\Users\\vwien\\dev\\workspace\\vsa_dashboard\\.gradle\\npm\\npm-v6.14.11\\node_modules\\npm\\bin\\npm-cli.js" "run" "test"
23 verbose node v14.15.0
24 verbose npm  v6.14.11
```

**Comment by atomfrede** [comment]
So the first thing is related to our gradle part, it seems not to resolve the placeholder from properties. The second thing you mention seems to be an npm/node problem. Will check. Can you set the node plugin version to `3.0.0-rc7`?

**Comment by vwiencek** [comment]
it's working fine with `3.0.0-rc7`

**Comment by vwiencek** [comment]
> 
> 
> it's working fine with `3.0.0-rc7`

I love it when a plan comes together!

**Comment by vwiencek** [comment]
Hello, for the first thing, the only solutions right now seem to be 
- harcode version number 
- or move back to version `2.2.4`

For the second point, I cannot reproduce it right now  .. do I need to clean something before relaunching `npm_run_test` ?

**Comment by wharleyinc** [comment]
> Hello, for the first thing, the only solutions right now seem to be
> 
> * harcode version number
> * or move back to version `2.2.4`
> 
> For the second point, I cannot reproduce it right now .. do I need to clean something before relaunching `npm_run_test` ?


I was trying to push my codes to heroku (first push), and I got the "Invalid tag name "extension 'node' property 'npmVersion'": error, using jhipster heroku.

How I resolved;
1. Removed package.json from C:\Users\CURRENT USER\AppData\Local\heroku
2. Updated heroku cli by downloading latest version
3. Manually fixed my build.gradle by removing  npmVersion = "${npmVersion}". (thanks to @atomfrede )
4. Ran jhipster heroku again and all ran fine.

PS: When I compiled locally in -dev mode, all ran fine. Seems an issue with -prod profile.
